### ISSUE TITLE
feat: unified/omnisearch telescope picker

### ISSUE THREAD

**Issue by leonnardo** [GustavEikaas/easy-dotnet.nvim#193]
First off, awesome job on the plugin!

I develop back-end services in .NET, running on Linux environments. I prefer to use Neovim in WSL instead of Visual Studio or Visual Studio Code.

However, one thing I miss is a unified/omnisearch feature, like in Visual Studio or Rider. It'd be great to have a "one picker to rule them all" that searches files within a solution/project, document symbols, and workspace symbols.

I've started working on this in my config but haven't finished it yet. 
Do you think this fits with your plugin's goals?

**Comment by GustavEikaas** [comment]
Maybe, im not sure I understand. Does fzf-lua and/or telescope not provide that already?

**Comment by leonnardo** [comment]
> Maybe, im not sure I understand. Does fzf-lua and/or telescope not provide that already?

It does, but in separate pickers. What I'm proposing is to have all results in the same picker.
For example: if I have a file `Example.cs` with a class `Example` in it and a method `firstExample` in this class and I open this "omnisearch" picker and type `exam`, it will bring all 3 results, not only the file (in case of find_files or git_files picker) or the symbols (in case of the lsp_dynamic_workspace_symbols picker)

**Comment by GustavEikaas** [comment]
Honestly seems like an amazing plugin in itself no? Do you plan to make it C# or roslyn specific?

**Comment by leonnardo** [comment]
I was planning to support only for dotnet projects, but looks like this is a good usecase for a telescope extension

**Comment by GustavEikaas** [comment]
Sounds like a great telescope plugin!

When getting results for classes would you use treesitter or LSP to get the results?

**Comment by GustavEikaas** [comment]
Just cleaning up old issues, reopen if you want to keep discussing

**Comment by leonnardo** [comment]
> Sounds like a great telescope plugin!
> 
> When getting results for classes would you use treesitter or LSP to get the results?

Sorry, I forgot to answer you. I was considering using LSP to get the results. Do you think Treesitter would be a better idea?

**Comment by GustavEikaas** [comment]
No worries. LSP would definitely give you the best and most correct results but it would make a pretty tight coupling to open buffers. If a user opens neovim but doesnt open any buffers then your search wouldnt be able to utilize any LSPs. Its definitely a tradeoff worth considering but I still think LSP is the way to go

**Comment by leonnardo** [comment]
What if the plugin detects the presence of `.sln` or `.csproj` in root or parent directories, similar to how [roslyn.nvim](https://github.com/seblj/roslyn.nvim) does to start the LSP client, but in neovim start event instead of the buffer enter event?

**Comment by GustavEikaas** [comment]
> What if the plugin detects the presence of `.sln` or `.csproj` in root or parent directories, similar to how [roslyn.nvim](https://github.com/seblj/roslyn.nvim) does to start the LSP client, but in neovim start event instead of the buffer en


I guess that would work but in that case you are planning on spawning an lsp server dedicated for symbol searching? My main concern would be performance issues. It would be best to rely on already running LSPs. Afaik roslyn doesnt start before BufEnter

**Comment by leonnardo** [comment]
> Afaik roslyn doesnt start before BufEnter

If we can change the Roslyn start event when the current working directory has `.csproj` or `.sln` files, that should fix your concern, right? The startup might be slower, but I think that's a fair tradeoff

**Comment by GustavEikaas** [comment]
Yeah thats a totally valid tradeoff. If roslyn starts before entering a buffer it would work well I think
